src/App.tsx

```diff
@@ -14,37 +15,98 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  ABC = 'ABC',
+  LENGTH = 'LENGTH',
+  DEFFAULT = 'DEFAULT',
+}
+
+type Props = {
+  sortField: SortType;
+  order: boolean;
+};
+
+function getPreparedGoods(goods: string[], { sortField, order }: Props) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ABC:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [order, setOrder] = useState(false);
+  const listOfGoods = getPreparedGoods(goodsFromServer, { sortField, order });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.ABC)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.ABC,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setOrder(!order)}
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !order,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || order) && (
+          <button
+            onClick={() => {
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here